### DISCUSSION TITLE
Asymetric Distances When Swapping Target & Observer

### DISCUSSION THREAD

**Discussion by JonSeamans** [Stellarium/stellarium#3539] [General]
Stellarium v. 23.1

I'm not complaining nor suggesting fixing this.
I happened to discover when you swap Moon/ Earth target / observer locations, the distances aren't the same at exactly the same time, e.g now.

- Swap Moon and geocentric Earth → Distance differs by ~30 kilometers (0.0074% larger when observer is the Earth).  
- Swap Pluto and geocentric Earth → Distances are the same to the display resolution.

Things doesn't look simple, and I've not dug deeper, but barring round-off decisions, seems like the distance calc should be exactly symmetric for any target and observer swaps.  Do you know why this wouldn't be the case? Have I missed something?

Thank you,
Jon

**Comment by gzotti** [comment]
Hmm... May have to do with (how we deal with) aberration correction. However, this just led to discovery of a bug. Ctrl-G (Goto selected) triggers a crash when aberration correction is switched off. (EDIT: I can no longer reproduce it...)

A weird thing that we must take into account is that apparently aberration is already included in the Lunar ephemeris. This could probably lead to some unexpected (indeed buggy) results :-(

**Reply by gzotti** [reply]
Confirmed. This is the aberration switch. When observing Moon from Earth (exactly this combination), there is no change concerning aberration for the Moon, regardless of the setting. Probably we should "counter-compute" aberration when switching it off now. In any case, Earth as seen from the Moon has aberration correction applied to its position, hence the slight difference.

**Comment by JonSeamans** [comment]
Just to confirm, the _correct_ distance should always include aberration.  Therefore, until Stellarium is revised, the _correct_ geocentric distance between the Earth and Moon should be determined using the Moon as the observer and with aberration enabled, correct?

**Reply by gzotti** [reply]
Probably. Or rather the other way round, use positions without aberration (which is an optical effect). But we don't have Lunar position without aberration.

There is also an additional tiny sub-arcsecond correction in Lunar longitude to compensate for the difference between figure centre (visible) and centre of gravity (ephemeris position).

But if you need reference data (apparently the topic is important for you), you should rather use JPL/Horizons. _Do not use Stellarium to control your spacecraft!_

**Comment by JonSeamans** [comment]
Is lunar position require aberration because it’s moving around the Earth?  In general, I would expect spacecraft positions require aberration too. Is that so?

Regarding my concern for small position changes, the truth of the matter is I’m not after a single true position but rather self consistency within ephemeris programs. In fact this whole thing started with an 80-km bodycentric-distance inconsistency fully within Horizons.  Whereas the same analysis conducted within Stellarium → 0.000000 km (with or without aberration correction). Note, Although close in magnitude, I’m not comparing absolute distances directly between Stellarium and Horizons. What bothers me is the distance difference should equal zero in both programs. 

Unfortunately, I can only compare results of Stellarium and Horizons. I don’t know the detailed calculations to do anything more.  If you’re interested, I can send you the specifics of this case. Yes, it does involve a spacecraft (Artemis / Orion) where I swap Orion and Earth as observer / target in both programs.  Unfortunately, Horizons isn’t working as a reference.

**Reply by gzotti** [reply]
Observers on the moving Earth need aberration correction applied for all sources of radiation they want to observe from outside Earth. However, it seems that the lunar ephemeris that we are using already includes aberration correction, while the positions of the planets must be corrected. I don't find this information in the accessible literature, though, so I cannot help you further.
If you find such discrepancies in Horizons, maybe they can answer your question. I'd be interested to hear from them.

**Comment by JonSeamans** [comment]
Oh, got it.  I didn't realize you were referring to Stellarium's current lack of lunar aberration correction.
Thanks for your added comments. I did send Horizons my question and some my results 5 days ago and not heard back yet. I'll ping him again and let you know if I learn anything new.

**Reply by gzotti** [reply]
Lunar position in Stellarium does include aberration. Always. Has done so since 2006 or so, unlike all other objects which are corrected for aberration only since 2021. Lunar position is in the "aberrated" (correct apparent) position even when aberration correction is switched off. This is probably because the ephemeris computations for the Moon include it out of the box. I just have no confirmation elsewhere why for the Moon no extra aberration correction should be applied. If I applied it like for any other object, the position would be wrong. Therefore I concluded it may have been built into the Lunar ephemeris.

**Reply by JonSeamans** [reply]
Huh, it looks to me like you've got the aberration status reversed. The attached tables suggest the (target) Moon distance from (observer) Earth is not corrected for aberration. Please check my numbers. Maybe I'm missing something.
![Stellarium - Earth,Moon Distance   Aberration Correction](https://github.com/Stellarium/stellarium/assets/75925212/27169300-a7af-420c-a1ae-e2e252ac1bd8)

**Reply by gzotti** [reply]
Then please explain why lunar POSITION matches other software titles and (more important) actual observations. I don't care for distance, this comes as side effect. Even if there is a mismatch in distance by almost 0.01%. 

(1) This is because there is no aberration shift for the Moon applied when observed from Earth (This and only this particular observer/target configuration). With this omission, position data matches reference solutions, occultations, etc. 
(2) This is because when observing from Moon, we are not observing Moon from Earth. 
(3) I cannot answer this question. Aberration is applied as geometrical shift of the observed object, thereby also influencing distance. 

(2) If this was true, my photos, GPS timing etc.  would be wrong. I trust my GPS timing.

**Reply by 10110111** [reply]
The computation of the aberration looks quite strange to me. For the planets & moons it's in `SolarSystem::computePositions`, right? What makes me wonder is that `aberrationPushSpeed` is unconditionally taken as the velocity of the observer's planet (as if wanting to compute aberration of the Sun), which makes no sense when we're trying to compute aberration resulting from relative velocities of the target and the observer, each being different not only by magnitude, but also by direction, since they all have different orbital phases.

**Reply by gzotti** [reply]
Here you have certainly seen my comment:

> // For higher accuracy, we now make two iterations of light time and aberration correction. In the final
> // round, we also compute rotation data.  May fix sub-arcsecond inaccuracies, and optionally apply
> // aberration in the way described in Explanatory Supplement (2013), 7.55.  For reasons unknown (See
> // discussion in GH:#1626) we do not add anything for the Moon when observed from Earth!  Presumably the
> // used ephemerides already provide aberration-corrected positions for the Moon?

The linked GH:#1626 has:

> Comparison against JPL HORIZONS is showing final deviations of typically 0.01s/0.1" only, in the present era. This is where I personally say "really good enough for my purposes". Historical values differ slightly also if compared in DT, probably from some different models for earth orientation.


You are invited to solve the remaining questions without breaking the observation match to photographs.

**Reply by 10110111** [reply]
> Here you have certainly seen my comment

I surely have. And it didn't answer my questions in any way. In both rounds the computation is identical (up to `updatePlanetCorrections`, whose implementation doesn't mention aberration). In both rounds aberration push is computed from observer planet's velocity and the distance to the target planet.

> You are invited to solve the remaining questions without breaking the observation match to photographs.

I'd like to first understand the logic already present, and then decide whether it's correct or broken, and only after this decide to fix it or leave untouched. Maybe I'm missing some place where planets' motion _relative to the observer_ is completed for the aberration push?
